### PR TITLE
feat: added '--skip-tag', '--skip-pr-check' and `--branch=*`

### PR DESCRIPTION
**Pull request by simlu**
This pr is not final. I'm putting this up since it solves my use case nicely. I do want to add more tests once we have agreed on changes here.

There are different reasons why options are added:

### skipTag
See discussion here: https://github.com/semantic-release/semantic-release/issues/753#issuecomment-455241656
Basically we can disable all plugin execution, however there is no way to prevent the tagging. So if we just want to output the version to a file without changing anything in the repo, this option is necessary.

### skipPrCheck
See discussion here: https://github.com/semantic-release/semantic-release/issues/1074
Basically CircleCI has weird behaviour. I can provide show case of the behaviour if that is desired.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -39,19 +39,21 @@ async function run(context, plugins) {
...
-  if (ciBranch !== options.branch) {
+  if (![ciBranch, '*'].includes(options.branch)) {
     logger.log(
       `This test run was triggered on the branch ${ciBranch}, while semantic-release is configured to only publish from ${
         options.branch
       }, therefore a new version won’t be published.`
     );
     return false;
   }
+  options.branch = isPr ? prBranch : ciBranch;
```
Comment (simlu) [inline index.js]: Ensure branch is correctly set for further use.

Comment (simlu) [inline index.js]: This should probably be moved to [L46](https://github.com/semantic-release/semantic-release/pull/1076/files#diff-168726dbe96b3ce427e7fedce31bb0bcR46)

File: index.js [hunk 2]
```diff
@@ -39,19 +39,21 @@ async function run(context, plugins) {
...
   }
 
-  if (isCi && isPr && !options.noCi) {
+  if (isCi && isPr && !options.noCi && !options.skipPrCheck) {
     logger.log("This run was triggered by a pull request and therefore a new version won't be published.");
     return false;
   }
 
-  if (ciBranch !== options.branch) {
+  if (![ciBranch, '*'].includes(options.branch)) {
```
Comment (simlu) [inline index.js]:
```suggestion
  if (![isPr ? prBranch : ciBranch, '*'].includes(options.branch)) {
```
Is probably better.

**Comment by pvdlg** [comment]
Before you spend more time on the PR, this is not something we will implement.

As mentioned in #753, outputting the version to be released without making a release is a feature that we are still evaluating.
So far the only use cases mentioned there that  are commons enough and explained in detail are supported via the pre-release feature in version 16.0.0-beta. This use case is to publish release for integration tests purpose only then once the test are successful to publish on the main release channel. This use case is detailed there: https://github.com/semantic-release/semantic-release/blob/beta/docs/recipes/pre-releases.md.
Other comments in #753 are either:
- of the type "I want it because that's my workflow" without explanation justifying why that workflow is better than the proposed alternative
- or are not common enough to justify the implementation of the feature

Regarding skipping the PR check this is not something we will implement as we do not want to make releases from a PRs for obvious reasons. If we fail to detect properly that a CI job has been triggered by PR, then will fix the detection.

**Comment by simlu** [comment]
@pvdlg Sure. I've mostly spend time on this since I'm now deploying using my fork. That works great for now.

**Comment by pongells** [comment]
> Before you spend more time on the PR, this is not something we will implement.
> 
> As mentioned in #753, outputting the version to be released without making a release is a feature that we are still evaluating.
> So far the only use cases mentioned there that are commons enough and explained in detail are supported via the pre-release feature in version 16.0.0-beta. This use case is to publish release for integration tests purpose only then once the test are successful to publish on the main release channel. This use case is detailed there: https://github.com/semantic-release/semantic-release/blob/beta/docs/recipes/pre-releases.md.
> Other comments in #753 are either:
> 
> * of the type "I want it because that's my workflow" without explanation justifying why that workflow is better than the proposed alternative
> * or are not common enough to justify the implementation of the feature
> 
> Regarding skipping the PR check this is not something we will implement as we do not want to make releases from a PRs for obvious reasons. If we fail to detect properly that a CI job has been triggered by PR, then will fix the detection.

I have a use case and a motivation for the `--skip-tag`. 

Use case: I want to use this module locally to generate the CHANGELOG and update the version(s) of my package(s). I want to be able to check the CHANGELOG and eventually manually modify it before merging to master for release. 

Motivation: we can customize and/or skip almost everything, I don't see why the tagging should be forced upon us when it would be so easy to just add the suggested flag.

**Comment by rplotkin** [comment]
So, please look at the git-flow and one-flow models. In these models, the release branch _is what is tagged_. That release branch undergoes QA review (ideally), and is merged from a PR. I have to work around the refusal to integrate --skip-pr-check despite our git pattern being one commonly utilized in enterprise.

**Comment by Ridermansb** [comment]
Yes.. I agree with @rplotkin .. I have the same issue with `--skip-pr-check`

**Comment by JoA-MoS** [comment]
Use Cases (any language / framework that requires the build number at build time) 
* C#
* Java / Gradle (gradle.properties)
* There are probably a lot more
* Bundling your app into a docker image and using semantic release to publish the docker image
